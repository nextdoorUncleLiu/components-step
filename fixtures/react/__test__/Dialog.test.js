import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import Dialog from '../Dialog'
import ComponentsStep from '../../../dist/main.js'
console.log(ComponentsStep)
function Mid1(props) {
  return () => {
    return (
      <div>mid1 {props ? 'true' : 'false'}</div>
    )
  }
}

function Mid2(props) {
  return () => {
    return (
      <div>mid2 {!props ? 'true' : 'false'}</div>
    )
  }
}

function HighFn(atom, attr) {
  return class extends Component {
    constructor() {
      super()
    }
    render() {
      return (
        <Dialog {...atom} />
      )
    }
  }
}
const Test1 = ComponentsStep([
  {
    label: 'Step1',
    stepItemMiddle: Mid1
  },
  {
    label: 'Step2',
    stepItemMiddle: Mid2
  }
])(HighFn, 1)
test('test container', () => {
  const component = renderer.create(
    <Test1 />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})