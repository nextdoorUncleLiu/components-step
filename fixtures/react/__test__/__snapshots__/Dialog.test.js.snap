// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test container 1`] = `
<div>
  <div>
    mid1 
    true
  </div>
  <div>
    <label>
      姓名
    </label>
    <input />
  </div>
  <div>
    <label>
      年龄
    </label>
    <input />
  </div>
  <div>
    mid2 
    false
  </div>
  <div>
    <button>
      确认
    </button>
  </div>
</div>
`;
